bin/dpipe.ml

```diff
@@ -32,87 +32,88 @@ let do_command input_size output_size mode level algo wbits =
   let dst =
     Bigarray.Array1.create Bigarray.Char Bigarray.c_layout output_size
   in
+  let refill dst len =
+    let count = unix_write B.bigstring Unix.stdout dst 0 len in
+    match count with
+      | -1 -> raise (Failure "Error occure: unix write returned -1")
```

Comment:
If we are really talking best practice, we should use `Unix.single_write` instead of `Unix.write` here; see https://ocaml.github.io/ocamlunix/ocamlunix.html#sec118

The idea is that some errors (`EINTR`; `EAGAIN`; `EWOULDBLOCK`, although the latter only applies to writes to `non-blocking` file descriptors) are recoverable (by trying again), but others (like `EINVAL`) do not.
Unfortunately writing to a file on UNIX-like systems is not easy, and requires somewhat careful reading of `man 2 write`. It is further complicated by the OCaml standard library's bindings to `libc` wrapping some things in exceptions in some cases.

This is completely unrelated to `GZip`, but I think that if we're providing example code it would be nice to clean it up.
If you don't want to get into this right now I'm happy to submit a PR once this PR has been merged, but if you want to give it a shot and get peer review I thought you should have the opportunity @TheFrenchTroll. :)

Reply:
I'm really interested by this, but I think I will let you this one. I need to see some foreign code to get used to the best methods before being able to claim mine as "clean" :wink: